### PR TITLE
Mobile assessment improvements

### LINKED ISSUE #2165: 'try again' obscured on mobile
In exercises, the mobile view obscures the user feedback:

image

### LINKED ISSUE #2618: improve UX of selecting a correct answer in exercises
### Observed behavior

A while back, we switched the 'check' button from primary to secondary. Looks like this happened here #1858 (though I don't see any record of it). @christianmemije and @jeepurs  mentioned the change had something to do with providing more positive feedback when a user gets an answer correct. (I.e. switching from secondary to primary was intended to be a form of user feedback).

However we've got new feedback that using a secondary button for 'check' diminishes the importance of it: https://github.com/fle-internal/kolibri-instant-schools-plugin/issues/125

image

The upshot is that we probably chose the wrong strategy for giving positive feedback after getting a correct answer.

### Expected behavior

* User should get positive feedback after getting an answer correct
* Checking an answer should be a glaringly obvious operation

### User-facing consequences

Some users are confused when checking their answers

(cc @khangmach @jtamiace)

### LINKED ISSUE #2630: assessment progress indicator has issues on mobile
### Observed behavior

Z-index is off, and bottom margin is missing:

image

### Expected behavior

cleaner layout

### PR DESCRIPTION
**Pull request by indirectlylit**
# Details

### Summary

fixes #2630, fixes #2165, fixes #2618

* increased z-index
* added conditional bottom margin
* add 'correct!' message when appropriate
* make 'check' button primary when 'next item' button is not visible
* fixed regression where first slot was supposed to be styled differently (thicker underline)

### Reviewer guidance

Please do some manual testing

# Contributor Checklist

- [x] PR has the correct target milestone
- [x] PR has the appropriate labels
- [ ] Changes in the PR do not introduce accessibility regressions ([confimed by testing with one of the recommended tools](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)) 
- [x] If PR is ready for review, it has been assigned or requests review from someone
- [ ] Documentation is updated as necessary
- [ ] External dependency files are updated (`yarn` and `pip`)
- [ ] If internal dependency is updated, link to diff is included
- [ ] Screenshots of any front-end changes are in the PR description
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] You've added yourself to AUTHORS.rst if you're not there

# Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR has been fully tested manually

### CODE REVIEW

**Comment by indirectlylit** [comment]
added some updates based on feedback in review today - ready for final review and merge

File: kolibri/plugins/learn/assets/src/views/content-page/index.vue [hunk 1]
```diff
@@ -39,6 +39,7 @@
       :channelId="channelId"
       :available="content.available"
       :extraFields="content.extra_fields"
+      :hasPrimaryButton="!showNextBtn"
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/content-page/index.vue]: ShowNextBtn returns an object so this does not work

localhost_8000_learn_

image

Comment (christianmemije) [reply kolibri/plugins/learn/assets/src/views/content-page/index.vue]: Or wait, does the screenshot show intentional behavior @indirectlylit ?

File: kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue [hunk 2]
```diff
@@ -39,7 +39,7 @@
     transition: transform $core-time ease-in
 
   .yes
-    fill: $core-action-normal
+    fill: $core-status-correct
 
   .yes.yay
     transform: scale(1.5)
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue]: White we are here, let's remove this unnecessary sizing that is done to the svgs

Comment (christianmemije) [reply kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue]: Or make it consistent for all SVGs

image

File: kolibri/plugins/learn/assets/src/views/content-page/index.vue (hunk 1 above)
Comment (indirectlylit) [reply kolibri/plugins/learn/assets/src/views/content-page/index.vue]: no, that's wrong. Only 'go to next item' should be primary.

File: kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue (hunk 2 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue]: that's intentional - it's supposed to be larger. Might be broken - I'll look into it

File: kolibri/plugins/learn/assets/src/views/content-page/index.vue (hunk 1 above)
Comment (indirectlylit) [reply kolibri/plugins/learn/assets/src/views/content-page/index.vue]: I don't understand how this is an object...

```javascript
      showNextBtn() {
        return this.progress >= 1 && this.content && this.nextContentLink;
      },
```

**Comment by christianmemije** [comment]
A few suggested style changes for assessment-wrapper:

* Remove all `font-size` css and instead just add `font-size: smaller` to `attemptprogress-container`. We do not change font-sizes for mobile in other places, so why here? Not necessary imo.
* For `attemptprogress-container.mobile`
  * Replace the border with a drop-shadow.
  * Change padding to 16px

File: kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue (hunk 2 above)
Comment (christianmemije) [reply kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue]: Well in that case, don't forget about the light bulb.

File: kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue [hunk 3]
```diff
@@ -140,6 +144,10 @@ oriented data synchronization.
         type: Function,
         default: () => Promise.resolve(),
       },
+      hasPrimaryButton: {
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue]: Can we make this prop more clear, e.g. theCheckButtonIsPrimary, well obviously shorter haha

File: kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue (hunk 2 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue]: lightbulbs count as wrong, so they were 'no'

Comment (christianmemije) [reply kolibri/core/assets/src/views/exercise-attempts/answer-icon.vue]: Got it.

File: kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue [hunk 4]
```diff
@@ -422,54 +430,42 @@ oriented data synchronization.
...
 
   .attemptprogress-container.mobile
+    border-top: thin solid $core-text-annotation
+    background-color: $core-bg-light
+    padding: 4px
     position: fixed
-    height: 60px
     width: 100%
-    border-radius: 0
+    height: 80px // if changed, also change `bottomSpaceReserved` in top-level index
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue]: Let's get rid of this set height and instead add
```
overflow-x: auto;
overflow-y: hidden;
```
to .wrapper to handle a large n

**Comment by jtamiace** [comment]
I used a hint and it did this... After I refreshed the page, it looked okay, and then I clicked "get a hint" again and did the same thing

screen shot 2017-11-17 at 4 56 34 pm

File: kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue (hunk 4 above)
Comment (indirectlylit) [reply kolibri/plugins/learn/assets/src/views/assessment-wrapper/index.vue]: those lines make sense, but I still need a fixed height, no?

**Comment by indirectlylit** [comment]
comments addressed - ready for another review

**Pull request merged**